lib/entities/quote/DutchQuote.ts

```diff
@@ -242,6 +261,19 @@ export class DutchQuote implements IQuote {
         recipient: this.request.config.swapper,
       });
 
+    if (
+      this.portionRecipient &&
+      this.portionBips &&
+      frontendAndUraEnablePortion(this.request.info.sendPortionEnabled)
+    ) {
+      builder.output({
+        token: this.tokenOut,
```

Comment:
Okay i am realizing now there is a bug here if i am not mistaken.

If we just add another output, without adjusting the first output or the amountIn, then this will fail since we create more outputTokens than what the MM promised.

Essentially, what we have to do is change like 259-260 to  `startAmount: this.amountOutStart - this. portionAmountOutStart` to account for the difference there.

Otherwise, the calldata encoding for this request would be off given that we would have created extra outputs.

cc: @marktoda @unicorn-here

Reply:
Similarly, for EXACT_OUT, the only way this would work is if portion is on, we broadcast to our MMs the EXACT_OUT value before the portion is applied.

In other words, if someone says they want a quote for 10 USDC exact_out, we should actually send to MM a request for 10.15. This way, we dont have to change the amountIn and adjust for it. All other solutions require adjusting amountIn based off the fees on the output token which would be hard.

Again tagging folks because I think this is crucial.

@marktoda @unicorn-here @rileydcampbell

Reply:
looks like this is resolved now